src/main/kotlin/io/unthrottled/doki/promotions/MotivatorPluginPromotionRunner.kt

```diff
@@ -40,17 +41,27 @@ class MotivatorPluginPromotion(
   }
 
   override fun run() {
-    ThemeManager.instance.currentTheme.ifPresent { dokiTheme ->
-      WindowManager.getInstance().suggestParentWindow(
-        ProjectManager.getInstance().openProjects.first()
-      ).toOptional()
-        .ifPresent {
-          MotivatorPromotionDialog(
-            dokiTheme, it, onPromotion
-          ).show()
-        }
-    }
-
+    MotivatorPluginPromotion.runPromotion(onPromotion)
     IdeEventQueue.getInstance().removeIdleListener(this)
   }
+}
+
+object MotivatorPluginPromotion {
+  fun runPromotion(onPromotion: (PromotionResults) -> Unit) {
+    ApplicationManager.getApplication().executeOnPooledThread {
+      ThemeManager.instance.currentTheme.ifPresent { dokiTheme ->
+        WindowManager.getInstance().suggestParentWindow(
+          ProjectManager.getInstance().openProjects.first()
+        ).toOptional()
+          .ifPresent {
+            val promotionAssets = PromotionAssets(dokiTheme)
+            ApplicationManager.getApplication().invokeLater {
+              MotivatorPromotionDialog(
```

Comment:
Once the work is done in the separate thread, we need the ability to show things in the UI. 
We can't trigger any UI classes on a thread that isn't the AWT event processing thread.

So the assets have been downloaded locally, then we just schedule the dialog to show up asynchronously on the UI thread. That's what `invokeLater` does.